lib/trino-filesystem-s3/src/test/java/io/trino/filesystem/s3/AbstractTestS3FileSystem.java

```diff
@@ -167,29 +170,86 @@ void testFileWithTrailingWhitespaceAgainstNativeClient()
     }
 
     @Test
-    void testExistingFileWithTrailingSlash()
+    void testExistingDirectoryWithTrailingSlash()
             throws IOException
     {
-        try (S3Client s3Client = createS3Client()) {
-            String key = "data/file/";
-            s3Client.putObject(request -> request.bucket(bucket()).key(key), RequestBody.empty());
+        try (S3Client s3Client = createS3Client(); Closer closer = Closer.create()) {
+            String key = "data/dir/";
+            createDirectory(closer, s3Client, key);
             try {
                 assertThat(fileSystem.listFiles(getRootLocation()).hasNext()).isFalse();
 
                 Location data = getRootLocation().appendPath("data/");
                 assertThat(fileSystem.listDirectories(getRootLocation())).containsExactly(data);
-                assertThat(fileSystem.listDirectories(data)).containsExactly(data.appendPath("file/"));
+                assertThat(fileSystem.listDirectories(data)).containsExactly(data.appendPath("dir/"));
 
-                // blobs ending in slash are invisible to S3FileSystem and will not be deleted
                 fileSystem.deleteDirectory(data);
-                assertThat(fileSystem.listDirectories(getRootLocation())).containsExactly(data);
+                assertThat(fileSystem.listDirectories(getRootLocation())).isEmpty();
 
                 fileSystem.deleteDirectory(getRootLocation());
-                assertThat(fileSystem.listDirectories(getRootLocation())).containsExactly(data);
+                assertThat(fileSystem.listDirectories(getRootLocation())).isEmpty();
             }
             finally {
```

Comment:
Sorry for the churn, but I just noticed that we don't need this anymore, since it will be cleaned up by `TempDirectory`. With that change, we can also inline `key`.